benchexec/tablegenerator/react-table/src/components/TaskDefinitionViewer.js

```diff
@@ -0,0 +1,114 @@
+  constructor(props) {
+    super(props);
+    this.state = {
+      splitterTag: "<splitter#9d81y23>",
+      fileTag: "<file#092nt43>",
+      content: this.props.yamlText,
+    };
+  }
+
+  componentDidMount() {
+    this.prepareTextForRendering();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.yamlText !== this.props.yamlText) {
+      this.prepareTextForRendering();
+    }
+  }
+
+  /**
+   * Parses the YAML file and encloses all input and property files with a fileTag as well as a splitterTag,
+   * so they can be rendered separately as links. Takes the following assumptions:
+   * input_files is either a string or a list of strings
+   * properties is a list of dicts, each with a "property_file" key
+   */
+  prepareTextForRendering = () => {
+    const yamlObj = yamlParser.parseDocument(this.props.yamlText);
+
+    const inputFiles = yamlObj.get("input_files");
+    if (inputFiles) {
+      if (Array.isArray(inputFiles.items)) {
+        inputFiles.items.forEach(inputFileItem => {
+          inputFileItem.value = this.encloseFileInTags(inputFileItem.value);
+        });
+      } else {
+        yamlObj.set("input_files", this.encloseFileInTags(inputFiles));
+      }
+    }
+
+    const properties = yamlObj.get("properties");
+    if (properties) {
+      if (Array.isArray(properties.items)) {
+        properties.items.forEach(property => {
+          if (Array.isArray(property.items)) {
+            property.items.forEach(propertyItem => {
+              if (propertyItem.key.value === "property_file") {
+                propertyItem.value.value = this.encloseFileInTags(
+                  propertyItem.value.value,
+                );
+              }
+            });
+          }
+        });
+      }
+    }
+
+    this.setState({ content: yamlObj.toString() });
+  };
+
+  encloseFileInTags = fileName => {
+    return (
+      this.state.splitterTag +
+      this.state.fileTag +
+      fileName +
+      this.state.fileTag +
+      this.state.splitterTag
+    );
+  };
+
+  render() {
+    const contentBySplitter = this.state.content.split(this.state.splitterTag);
+    const jsxContent = [];
+
+    contentBySplitter.forEach(contentPart => {
+      let jsxContentPart;
+      // If contentPart is enclosed with file tags (= if contentPart is a file which should be linked)
+      if (
+        contentPart.match(
+          "^" + this.state.fileTag + "(?:.)+" + this.state.fileTag + "$",
+        )
+      ) {
+        contentPart = contentPart.replace(
+          new RegExp(this.state.fileTag, "g"),
+          "",
+        );
+        jsxContentPart = (
+          <span
+            onClick={() => this.props.loadNewFile(contentPart)}
+            className="link-overlay-file-link"
+            key={contentPart}
+          >
+            {contentPart}
+          </span>
+        );
+      } else {
+        jsxContentPart = <span key={contentPart}>{contentPart}</span>;
```

Comment:
Wouldn't it be possible to skip the attribute-less `<span>`?
```suggestion
        jsxContentPart = contentPart;
```

Reply:
I think the issue here is that we will get a warning if we render a list of items without giving each item a key.

Reply:
Even if those items without a key are just plain strings?

Reply:
Hmmm good point. 
I rarely render a list of plain strings in react and just assumed that there is some internal wrapping into JSX objects going on. 
I tested it in a code-sandbox and can verify that it does not care about plain strings not having keys. 
Disregard my previous comment.

Reply:
The point of using the `<span>` tag here was indeed solely for giving a key to the items. I've also just tested changing them to plain strings and didn't notice any unwanted behavior or warnings, so there should be no problem removing those tags.